app/lib/search/text_utils.dart

```diff
@@ -33,3 +34,9 @@ Iterable<String> splitForIndexing(String text) {
   if (text == null || text.isEmpty) return new Iterable.empty();
   return text.split(_separators).where((s) => s.isNotEmpty);
 }
+
+List<String> extractExactPhrases(String text) => _exactTermRegExp
+    .allMatches(text)
+    .map((m) => m.group(1).trim())
+    .where((s) => s.isNotEmpty)
+    .toList();
```

Comment:
Add a `.map(normalizeTextBeforeIndexing)` here and remove it on the usage site (which does it repeatedly atm).

Reply:
I've put the `.map` at the usage point, it keeps the extraction and the indexing/matching of it separate.